store/snapshots/manager.go

```diff
@@ -165,6 +165,7 @@ func (m *Manager) Create(height uint64) (*types.Snapshot, error) {
 		return nil, errorsmod.Wrap(storetypes.ErrLogic, "no snapshot store configured")
 	}
 
+	m.multistore.AnnounceSnapshotHeight(int64(height))
```

Comment:
if we have a `*WithAnnounce` interface, we can just check if the store supports this method (by casting) and call it if so

Reply:
Without the announcement call, the memory leak on restart is still possible. But I agree that this is a good way to preserve backwards compatibility. I have pushed the changes